libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +250,133 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
         NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
 }
 
+// calculate poi information
+void AP_Camera_Backend::calculate_poi()
+{
+#if AP_TERRAIN_AVAILABLE
+    if (!poi_calculation.sem.take_nonblocking()) {
+        return;
+    }
+    const AP_AHRS &ahrs = AP::ahrs();
+    if (!ahrs.get_location(poi_calculation.loc)) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: vehicle pos unavailable");
+        return;
+    }
+
+    // change vehicle location to ASML
+    poi_calculation.loc.change_alt_frame(Location::AltFrame::ABSOLUTE);
+
+    // retrieve gimbal attitude
+    auto *mount = AP_Mount::get_singleton();
+    Vector3f att;
+    if (!mount->get_attitude_euler(get_mount_instance(),att[0], att[1], att[2]) || isnan(att[1]) || isnan(att[2])) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: gimbal attitude unavailable");
+    }
+
+    // project forward from vehicle looking for terrain
+    // start testing at vehicle's location
+    Location test_loc = poi_calculation.loc, prev_test_loc = poi_calculation.loc;
+
+    // get terrain altitude (asml) at test_loc
+    bool success = false;
+    auto terrain = AP_Terrain::get_singleton();
+    float terrain_amsl_m, prev_terrain_amsl_m;
+    success = terrain->height_amsl(test_loc, terrain_amsl_m, true);
```

Comment:
```suggestion
    const bool success = terrain->height_amsl(test_loc, terrain_amsl_m, true);
```


... but can probably do without the variable altogether